std/math.d

```diff
@@ -522,7 +522,10 @@ real abs(Num)(Num y) @safe pure nothrow @nogc
     assert(abs(71.6Li) == 71.6L);
     assert(abs(-56) == 56);
     assert(abs(2321312L)  == 2321312L);
-    assert(abs(-1+1i) == sqrt(2.0L));
+    assert(abs(creal(-1+1i)) == sqrt(2.0L));
+    cdouble c = -1+1i;
+    double d = 2.0;
+    assert(abs(c) == sqrt(d));
```

Comment:
Sure, but I expect if you use a debugger to look at the register contents they'll tell a different story.